frontend/src/components/HeaderUserMenu.js

```diff
@@ -33,8 +34,17 @@ UserMenuNav.propTypes = {
 };
 
 function HeaderUserMenu() {
+  let hasImpersonationKey;
+  try {
+    hasImpersonationKey = window.localStorage.getItem(LOCAL_STORAGE_IMPERSONATION_KEY) !== null;
```

Comment:
this is probably fine, but we do have a ```storageAvailable``` function in ```/hooks/helpers``` as well (just as an FYI)

Is it possible that session storage is a better choice for this? Impersonation for the life of a browser tab, that way you could be accessing the site as two users at once? Not sure how we'll use this 100% so not sure how to properly weigh costs/benefits to both

Reply:
the other weird corner case is thing is for users, only people on our team as far as I can think of, who use the site in multiple environments, I think session storage is better in that scenario as well

Reply:
this is a great idea! I made this change here: bc9cae8